client/src/js/services/grid/TransactionService.js

```diff
@@ -303,24 +303,81 @@ function TransactionService($timeout, util, uiGridConstants, bhConstants, Notify
    * This method is used to check if a transaction is balanced by their record
    */
   function validateTransaction(entity) {
-    var transaction = entity.data.data;
-    var debit = 0,
-      credit = 0;
-    
-    transaction.forEach(function (row) {
-      debit += Number(row.debit_equiv);
-      credit += Number(row.credit_equiv);
+    var transaction = [];
+
+    // include transaction data
+    entity.data.data.forEach(function (row) {
+      transaction.push(row);
     });
 
-    var ERR_UNBALANCED_TXN = 'POSTING_JOURNAL.ERRORS.UNBALANCED_TRANSACTIONS';
+    // include new rows
+    entity.newRows.data.forEach(function (row) {
+      transaction.push(row);
+    });
 
-    // later in validateTransaction()
+    var numberOfLine = transaction.length;
     var error;
 
-    if (debit !== credit) {
-      error = ERR_UNBALANCED_TXN;
+    // If the transaction are single line transaction 
+    if(numberOfLine === 1){
+      error = 'POSTING_JOURNAL.ERRORS.SINGLE_LINE_TRANSACTION';
+    } else {
+      var debit = 0,
+        credit = 0,
+        initialDate,
+        accountNull = false,
+        dateNull =  false,
+        dateDifferent = false,
+        dateWrongPeriod = false;
+
+      if(transaction[0].trans_date){
+        initialDate = transaction[0].trans_date; 
+      }
+        
+
+      transaction.forEach(function (row) {
+        debit += Number(row.debit_equiv);
+        credit += Number(row.credit_equiv);
+
+
+        // Check if they are account number Null
+        if(!row.account_number){
+          accountNull = true;
+        }
+
+        // Check if they are trans_date Null
+        if(!row.trans_date){
+          dateNull = true;
+        }
+
+        // Check if they are different Date 
+        if(Dates.util.str(row.trans_date) !== Dates.util.str(initialDate)){
+          dateDifferent = true;
+        }
+
+        // check if trans_date is in bed period
+        if(new Date(row.trans_date) < new Date(row.period_start) || new Date(row.trans_date) > new Date(row.period_end)){
+          dateWrongPeriod = true;
+        }
+      });
+
+      if(accountNull) {
+        error = 'POSTING_JOURNAL.ERRORS.MISSING_ACCOUNTS';
```

Comment:
Instead of having these long text strings here, it would be nicer to list them all at the top of the module, like is done with `ROW_EDIT_FLAG` and others at the top of this file (https://github.com/IMA-WorldHealth/bhima-2.X/pull/1501/files#diff-2b8e7d8131d2079c84c7b41baa2361a5R25).  Something like this:
```js
var ERR_MISSING_ACCOUNTS = 'POSTING_JOURNAL.ERRORS.MISSING_ACCOUNTS';
var ERR_MISSING_DATES = 'POSTING_JOURNAL.ERRORS.MISSING_DATES';
//....

// much later...

if (accountNull) {
 error = ERR_MISSING_ACCOUNTS;
} else if { /// etc...
```

This will make it much easier to update translation keys in the future - ideally these keys will be shared with the complex vouchers module.